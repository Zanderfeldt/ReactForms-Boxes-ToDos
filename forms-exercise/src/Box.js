import React from "react";

const Box = ({id, removeBox, color = 'blue', width = 5, height = 5}) => {
  const remove = () => removeBox(id);
  return (
    <div>
      <div 
        className="Box"
        style={{
          width: `${width}em`,
          height: `${height}em`,
          backgroundColor: `${color}`
        }}/>
        <button onClick={remove}>X</button>
    </div>
  );
}

export default Box