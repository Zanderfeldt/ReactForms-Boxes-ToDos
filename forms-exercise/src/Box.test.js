import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import Box from './Box';

it("render without crashing", () => {
  render(<Box />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});