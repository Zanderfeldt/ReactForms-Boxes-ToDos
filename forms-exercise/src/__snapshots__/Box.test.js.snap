// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Box"
      style="width: 5em; height: 5em; background-color: blue;"
    />
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
